### PR TITLE
[FIX] mrp: correct workorder quantities on backorders

### PR DESCRIPTION
**Pull request by HANNICHE-Walid**
Steps to reprodue:

-In settings enablle workorders and quality checks
-Create a BOM with multiple operations each having a quality check 
-Create an MO for that BOM with quantity set to 10 
-Pass the quality checks for all the workorders execpt the last 
-On the last WO change the quality and pass only for 3 
-Mark MO as done and Create a backorder for the remaining 7

Bug:
probably a typo
the already done quantities on all the workorders beyond the first are ignored (index // workorders_len is always 0)

Fix:
back port of this fix[1] also remving unused variable (initial_qty)
unused since[2]

[1]:https://github.com/odoo/odoo/commit/19f922920ed512213eb8d844537282ab4125e337

[2]:https://github.com/odoo/odoo/commit/5add23fb921d9e00920f8d1fd6c37b7cac59ef60

opw-3162603

### CODE REVIEW

**Comment by HANNICHE-Walid** [comment]
@fw-bot up to saas-15.2

**Comment by ajf-odoo** [comment]
lgtm
As all productions (initial & backorders) have the same amount of workorders, I would rename prod_workorders_len to workorders_len 
Also make a reference for a backport of #84502 as most of your code comes from it

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -1713,12 +1714,22 @@ def _default_amounts(production):
...
                     initial_workorder_remaining_qty[index % workorders_len] = max(remaining_qty - workorder.qty_produced, 0)
+                    if not workorder.qty_remaining:
+                        workorders_to_cancel += workorder
+                    elif workorders_to_update[-1:].production_id != workorder.production_id:
+                        workorders_to_update += workorder
+                elif workorder.qty_remaining and workorders_to_update[-1:].production_id != workorder.production_id:
+                    workorders_to_update += workorder
+
+        workorders_to_cancel.action_cancel()
+        for workorder in workorders_to_update:
```
Comment (Whenrow) [inline addons/mrp/models/mrp_production.py:1731]: For performance reason, I would set the state as follows:
```python
workorders.filtered(lambda wo: wo.next_work_order_id.production_availability == 'assigned').state = 'ready' 
workorders.filtered(lambda wo: wo.next_work_order_id.production_availability != 'assigned').state = 'waiting'
```

Comment (HANNICHE-Walid) [reply addons/mrp/models/mrp_production.py]: good point 
done :)

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**